apollo-router/src/services/layers/query_analysis.rs

```diff
@@ -105,28 +107,46 @@ impl QueryAnalysisLayer {
             .query
             .clone()
             .expect("query presence was already checked");
-        let entry = self
-            .cache
-            .lock()
-            .await
-            .get(&QueryAnalysisKey {
-                query: query.clone(),
-                operation_name: op_name.clone(),
-            })
-            .cloned();
+
+        let entry = {
+            let mut cache = self.cache.lock().unwrap();
+            cache
+                .get(&QueryAnalysisKey {
+                    query: query.clone(),
+                    operation_name: op_name.clone(),
+                })
+                .cloned()
+        };
 
         let res = match entry {
             None => {
                 let span = tracing::info_span!("parse_query", "otel.kind" = "INTERNAL");
-                match span.in_scope(|| self.parse_document(&query, op_name.as_deref())) {
+
+                let schema = self.schema.clone();
+                let conf = self.configuration.clone();
+                let q = query.clone();
+                let op = op_name.clone();
+
+                let result = task::spawn_blocking(move || {
+                    Query::parse_document(q.as_str(), op.as_deref(), schema.as_ref(), conf.as_ref())
+                })
+                .instrument(span)
+                .await
+                .unwrap();
```

Comment:
Bringing attention to this `unwrap()`. This happens if the task panics, I don't think we want to recover from that.

Reply:
this should be fine, yes. Could you replace with an `expect()` with a message explaining the rationale?